kotsadm/operator/pkg/client/deploy.go

```diff
@@ -54,17 +54,29 @@ func (c *Client) diffAndRemovePreviousManifests(applicationManifests Application
 	}
 
 	// we need to find the gvk+names that are present in the previous, but not in the current and then remove them
+	// namespaces that were remoed from YAML and added to additionalNamespaces should not be removed
 	decodedPreviousStrings := strings.Split(string(decodedPrevious), "\n---\n")
 	decodedPreviousMap := map[string]string{}
+
+DecodedPrevious:
 	for _, decodedPreviousString := range decodedPreviousStrings {
-		decodedPreviousMap[GetGVKWithNameAndNs([]byte(decodedPreviousString), targetNamespace)] = decodedPreviousString
+		k, o := GetGVKWithNameAndNs([]byte(decodedPreviousString), targetNamespace)
+		if o.APIVersion == "v1" && o.Kind == "Namespace" {
+			for _, n := range applicationManifests.AdditionalNamespaces {
```

Comment:
why can't this logic be part of this loop? https://github.com/replicatedhq/kots/pull/1620/files#diff-b8923687b620645f0bb41062a607692033e9a2d13f79a2893e8c1da91b16cf06R97